make/hotspot/lib/JvmOverrideFiles.gmk

```diff
@@ -52,16 +52,14 @@ ifneq ($(FDLIBM_CFLAGS), )
 endif
 
 ifeq ($(call isTargetOs, linux), true)
-  BUILD_LIBJVM_sharedRuntimeTrig.cpp_CXXFLAGS := -DNO_PCH $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
-  BUILD_LIBJVM_sharedRuntimeTrans.cpp_CXXFLAGS := -DNO_PCH $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
+  BUILD_LIBJVM_sharedRuntimeTrig.cpp_CXXFLAGS := $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
+  BUILD_LIBJVM_sharedRuntimeTrans.cpp_CXXFLAGS := $(FDLIBM_CFLAGS) $(LIBJVM_FDLIBM_COPY_OPT_FLAG)
 
-  ifeq ($(TOOLCHAIN_TYPE), clang)
-    JVM_PRECOMPILED_HEADER_EXCLUDE := \
-	sharedRuntimeTrig.cpp \
-	sharedRuntimeTrans.cpp \
-        $(OPT_SPEED_SRC) \
-	#
-  endif
+  JVM_PRECOMPILED_HEADER_EXCLUDE := \
+      sharedRuntimeTrig.cpp \
+      sharedRuntimeTrans.cpp \
+      $(OPT_SPEED_SRC) \
```

Comment:
Note that OPT_SPEED_SRC contains a rather large set of files. They have previously been compiling using PCH just fine with gcc. I don't think it is motivated to exclude them everywhere, just because gcc on ppc64 has a problem.

Reply:
> Note that OPT_SPEED_SRC contains a rather large set of files. They have previously been compiling using PCH just fine with gcc. I don't think it is motivated to exclude them everywhere, just because gcc on ppc64 has a problem.

It seems usually it is empty, because in 99% of cases  people will not set the opt-size jvm feature 

```
193 ifeq ($(call check-jvm-feature, opt-size), true)
194   JVM_OPTIMIZATION := SIZE
195   OPT_SPEED_SRC := \
196       allocation.cpp \
```

I haven't checked what happens with those recent PCH changes, when opt-size is configured.  If it works I am fine with your approach too that does not  add those to  JVM_PRECOMPILED_HEADER_EXCLUDE .

Reply:
I must admit I did not realize this list was dependent on the `opt-size` feature, so I cannot say if my patch really works with that enabled. Maybe the OPT_SPEED_SRC list is needed then, after all.

Nevertheless, I think we can limit the change for ppc on gcc. When I added the -W flag for more detailed warnings, I was taught that it was the value of some -mpower-..somethingsomething flag that was problematic for PCHs. So it seems like it could be a very much PPC specific problem.

Reply:
> Nevertheless, I think we can limit the change for ppc on gcc.

That's fine. Do you want to do it in your PR or should I do it  in this one?

Reply:
I think it is better to do this change separately here; I intend to remove the corresponding changes in my PR.

Did you test what happened when opt-size was configured, if you needed to exclude all the files in OPT_SPEED_SRC  as well?

Reply:
> I think it is better to do this change separately here; I intend to remove the corresponding changes in my PR.
> 

Good, let's do it this way.  I added one more commit .

> Did you test what happened when opt-size was configured, if you needed to exclude all the files in OPT_SPEED_SRC as well?

Yes I just did ;  we need the  OPT_SPEED_SRC file list added, otherwise on Linuxppc64le we fail when configuring opt-size .